lib/internal/streams/operators.js

```diff
@@ -216,21 +214,15 @@ async function some(fn, options) {
 }
 
 async function every(fn, options) {
-  if (typeof fn !== 'function') {
-    throw new ERR_INVALID_ARG_TYPE(
-      'fn', ['Function', 'AsyncFunction'], fn);
```

Comment:
@benjamingr Do you have an opinion on this?

Reply:
actually it kind of is since it's important that users know they can pass a function returning a promise

Reply:
Maybe we should have a `validateAsyncFunction` validator? This would require more work to find instances in the code base where we use `validateFunction` when the API accepts promises, maybe it shouldn't block this PR.

Reply:
Is it really necessary though? The validator telling the users to pass a function can also imply that they can also pass an async function, and in most places we just use the `validateFunction()` validator either way.

If it's actually necessary to indicate that, I can revert the changes to those.

Reply:
@VoltrexMaster it's not necessary it's just better. The user benefits from an error telling them they can pass an AsyncFunction since it implies the promise returned is useful to the function itself.